src/main/java/com/google/cloud/teleport/newrelic/dtos/coders/NewRelicLogRecordCoder.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.newrelic.dtos.coders;
+
+import com.google.cloud.teleport.newrelic.dtos.NewRelicLogRecord;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.apache.beam.sdk.coders.AtomicCoder;
+import org.apache.beam.sdk.coders.BigEndianLongCoder;
+import org.apache.beam.sdk.coders.NullableCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * A {@link org.apache.beam.sdk.coders.Coder} for {@link NewRelicLogRecord} objects. It allows
+ * serializing and deserializing {@link NewRelicLogRecord} objects, which can then be transmitted
+ * through a Beam pipeline using PCollection/PCollectionTuple objects.
+ */
+public class NewRelicLogRecordCoder extends AtomicCoder<NewRelicLogRecord> {
```

Comment:
See comments for NewRelicLogApiSendErrorCoder. The same applies for this one too.